.drone.yml

```diff
@@ -98,6 +98,101 @@ volumes:
   - name: dockertmp
     temp: {}
 
+---
+kind: pipeline
+type: kubernetes
+name: post-merge build
+
+trigger:
+  event:
+  - push
+  branch:
+  - master
+
+steps:
+  - name: short circuit docs changes
+    image: docker:git
+    commands:
+      # If a change is entirely documentation, skip the expensive build & test steps.
+      # https://discourse.drone.io/t/how-to-exit-a-pipeline-early-without-failing/3951
+      - ./build.assets/drone/diff-is-all-docs.sh $DRONE_COMMIT_BEFORE..$DRONE_COMMIT_AFTER && exit 78 || exit 0
+  - name: fetch tags
+    image: docker:git
+    commands:
+      - git fetch --tags
+  - name: wait for docker
+    image: docker
+    commands:
+      - timeout 15s /bin/sh -c 'while [ ! -S /var/run/docker.sock ]; do sleep 1; done'
+      - docker version
+    volumes:
+      - name: dockersock
+        path: /var/run
+  - name: build
+    image: docker:git
+    environment:
+      GITHUB_RO_KEY_PR:
+        from_secret: GITHUB_RO_KEY_PR
+      AWS_ACCESS_KEY_ID:
+        from_secret: AWS_S3_RO_ACCESS_KEY_ID
+      AWS_SECRET_ACCESS_KEY:
+        from_secret: AWS_S3_RO_SECRET_ACCESS_KEY
+    commands:
+      - apk add --no-cache make bash libc6-compat aws-cli
+      - mkdir -m 0700 /root/.ssh && echo "$GITHUB_RO_KEY_PR" > /root/.ssh/id_ed25519 && chmod 600 /root/.ssh/id_ed25519
+      - ssh-keyscan -H github.com > /root/.ssh/known_hosts 2>/dev/null && chmod 600 /root/.ssh/known_hosts
+      - make -C e production telekube opscenter
```

Comment:
Might to be explicit about deleting the SSH key from the storage once the operation it's used for has completed, although I presume this is effectively done already because the container terminates after this step and the key isn't written to the shared workspace?

Reply:
This key is conditionally used for a variety of clones of public repos.
For example:

https://github.com/gravitational/gravity/blob/3cb2505cf2a5fc7f2bb23114e8d6c7c4b1193f3d/build.assets/Makefile#L697
https://github.com/gravitational/gravity/blob/3cb2505cf2a5fc7f2bb23114e8d6c7c4b1193f3d/build.assets/Makefile#L168

> I presume this is effectively done already because the container terminates after this step and the key isn't written to the shared workspace?

Yeah, I was implicitly leveraging the transient nature of /root to scope the key to just this container.  Even if we deleted it from the FS, it would still be present in the environment -- though I suppose I could unset it there too.

Unlike some of the other credentials, this key doesn't have access to any private repos -- thus it is quite low sensitivity.  I needed it because I didn't want to rewite a bunch of build scripts to use public https clones instead of git clones.